### PR TITLE
test: skip test on rootless cgroupsv1

### LINKED ISSUE #11785: cgroupsv1(?): cannot toggle freezer: cgroups not configured for container
Almost certainly related to #11784, but this one is rootless-only. Seen in [f33](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6422964389281792/html/sys-podman-fedora-33-rootless-host.log.html#t--00236). This is blocking #11776.
```
[+0926s] not ok 236 podman selinux: shared context in (some) namespaces
         # (from function `is' in file test/system/helpers.bash, line 508,
         #  in test file test/system/410-selinux.bats, line 126)
         #   `is "$output" "$context_c1" "new container, run with --pid of existing one "' failed
         # $ podman rm --all --force
         # $ podman ps --all --external --format {{.ID}} {{.Names}}
         # $ podman images --all --format {{.Repository}}:{{.Tag}} {{.ID}}
         # quay.io/libpod/testimage:20210610 9f9ec7f2fdef
         # $ podman run -d --name myctr quay.io/libpod/testimage:20210610 top
         # 3aa057fa598b4cfd220e66e782944b01f59c00afbc7e41a2b4374d0249fa7ad3
         # $ podman exec myctr cat -v /proc/self/attr/current
         # system_u:system_r:container_t:s0:c226,c233^@
         # $ podman run --name myctr2 --ipc container:myctr quay.io/libpod/testimage:20210610 cat -v /proc/self/attr/current
         # system_u:system_r:container_t:s0:c226,c233^@
         # $ podman run --rm --pid container:myctr quay.io/libpod/testimage:20210610 cat -v /proc/self/attr/current
         # system_u:system_r:container_t:s0:c226,c233^@time="2021-09-28T17:10:04-05:00" level=warning msg="cannot toggle freezer: cgroups not configured for container"
         # time="2021-09-28T17:10:04-05:00" level=warning msg="cannot toggle freezer: cgroups not configured for container"
         # time="2021-09-28T17:10:04-05:00" level=warning msg="lstat : no such file or directory"
         # #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
         # #|     FAIL: new container, run with --pid of existing one 
         # #| expected: 'system_u:system_r:container_t:s0:c226,c233^@'
         # #|   actual: 'system_u:system_r:container_t:s0:c226,c233^@time="2021-09-28T17:10:04-05:00" level=warning msg="cannot toggle freezer: cgroups not configured for container"'
         # #|         > 'time="2021-09-28T17:10:04-05:00" level=warning msg="cannot toggle freezer: cgroups not configured for container"'
         # #|         > 'time="2021-09-28T17:10:04-05:00" level=warning msg="lstat : no such file or directory"'
         # #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
As with #11784, I see hundreds of instances of this in CI logs but cannot reproduce it myself on a cgroupsv1 f33 host. While trying to reproduce it, though, I did get a different error:
```
$ while :;do bats --filter namespaces /usr/share/podman/test/system/410-selinux.bats || break;done
 ✗ podman selinux: shared context in (some) namespaces
   (from function `die' in file /usr/share/podman/test/system/helpers.bash, line 448,
    from function `run_podman' in file /usr/share/podman/test/system/helpers.bash, line 221,
    in test file /usr/share/podman/test/system/410-selinux.bats, line 125)
     `run_podman run --rm --pid container:myctr $IMAGE cat -v /proc/self/attr/current' failed with status 126
   $ podman rm --all --force
   $ podman ps --all --external --format {{.ID}} {{.Names}}
   $ podman images --all --format {{.Repository}}:{{.Tag}} {{.ID}}
   quay.io/libpod/testimage:20210610 9f9ec7f2fdef
   $ podman run -d --name myctr quay.io/libpod/testimage:20210610 top
   2b49db4f018fd2fa275a93d0a70efbf304bd9d7f25715b91940cc8a57081b25f
   $ podman exec myctr cat -v /proc/self/attr/current
   system_u:system_r:container_t:s0:c217,c675^@
   $ podman run --name myctr2 --ipc container:myctr quay.io/libpod/testimage:20210610 cat -v /proc/self/attr/current
   system_u:system_r:container_t:s0:c217,c675^@
   $ podman run --rm --pid container:myctr quay.io/libpod/testimage:20210610 cat -v /proc/self/attr/current
   Error: readlink: Permission denied: OCI permission denied
   [ rc=126 (** EXPECTED 0 **) ]
```

### PR DESCRIPTION
**Pull request by giuseppe**
skip the test "podman selinux: shared context in (some) namespaces" on
cgroupsv1 when running as rootless since the tests requires
--pid=container:.

If the container runtime cannot use cgroupsv1 and the container has no
pid namespace. then it is not possible to correctly terminate the
container.  Without a cgroup or a pid namespace, the runtime has no
control on what processes are in the container.

Closes: https://github.com/containers/podman/issues/11785

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

#### What this PR does / why we need it:

fix a failing test on rootless and cgroups v1

#### How to verify it

run the test on cgroups v1 and rootless

#### Which issue(s) this PR fixes:

Fixes #11785

### CODE REVIEW

**Comment by Luap99** [comment]
Should the podman run/create code prevent the user from doing this?

File: test/system/helpers.bash [hunk 1]
```diff
@@ -428,6 +428,18 @@ function skip_if_cgroupsv1() {
...
 }
 
+######################
+#  skip_if_rootless_cgroupsv1  #  ...with an optional message
+######################
+function skip_if_rootless_cgroupsv1() {
+    if is_rootless; then
+        if ! is_cgroupsv2; then
+            local msg=$(_add_label_if_missing "$1" "rootless cgroupvs1")
+            skip "${msg:-not applicable under rootless podman}"
```
Comment (edsantiago) [inline test/system/helpers.bash:438]: This is misleading; it should include something about cgroupsv1 as well. Perhaps:
```
    ...not supported as rootless under cgroupsv1}"
```

**Review by edsantiago — COMMENTED** [review]
Thank you for adding a helper! Could you also use it in https://github.com/containers/podman/blob/cd10304dca72ef030b64142885518e6dc0d3e4af/test/system/420-cgroups.bats#L11-L13

**Comment by giuseppe** [comment]
> Should the podman run/create code prevent the user from doing this?

It should be the OCI runtime to raise an error since it creates the cgroup, and if it fails then it can check if there is a PID namespace.

I am not sure though if we block valid use cases doing so., the worse that can happen is to leak processes in the initial container PID namespace

**Comment by edsantiago** [comment]
LGTM, thank you!

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**